python/pipeline/meso.py

```diff
@@ -519,55 +565,301 @@ def make(self, key):
+        # Run postprocessing on assumed xy shifts
+        x_shifts, y_shifts, outliers = galvo_corrections.process_xy_motion(x_shifts, y_shifts, key)
+
+        # Run second iteration of motion correction using a global template
+        if key["motion_correction_method"] in (
+            5,
+            7,
+        ):
+
+            print("Running second iteration...")
+
+            scan = scan[0, :, :, 0, :]
+            scan = scan.astype(np.dtype("float32"))
+
+            old_x_shifts = x_shifts.copy()
+            old_y_shifts = y_shifts.copy()
+
+            # Correct scan in place using low memory chunks
+            scan = galvo_corrections.low_memory_motion_correction(
+                scan, raster_phase, fill_fraction, x_shifts, y_shifts
+            )
+
+            # Provide a small amount of smoothing in time
+            if key["motion_correction_method"] in (5,):
+                window_size = 3
+            elif key["motion_correction_method"] in (7,):
+                fps = (ScanInfo & key).fetch1("fps")
+                window_size = np.max((int(fps / 5), 3))
+            else:
+                raise PipelineException(f"The smoothing for the second iteration of motion correction is not defined for {key['motion_correction_method']}")
+
+            # In place low memory convolution. This is slower
+            # but allows for the entire scan to be in memory.
+            start_indices = np.arange(0, scan.shape[-1], 1000)
+            if start_indices[-1] != scan.shape[-1]:
+                start_indices = np.insert(
+                    start_indices, len(start_indices), scan.shape[-1]
+                )
+            for start_idx, end_idx in tqdm(
+                zip(start_indices[:-1], start_indices[1:]), total=len(start_indices) - 1
+            ):
+                conv_start = np.max((0, start_idx - window_size))
+                conv_end = np.min((end_idx + window_size, scan.shape[-1]))
+                conv_fragment = ndimage.convolve1d(
+                    scan[:, :, conv_start:conv_end],
+                    np.ones(window_size) / window_size,
+                    axis=-1,
+                    mode="reflect",
+                )
+
+                # Use these so the offset will be zero when at beginning or end of scan
+                slice_start = start_idx - conv_start
+                slice_end = end_idx - conv_end + conv_fragment.shape[-1]
+                scan[:, :, start_idx:end_idx] = conv_fragment[
+                    :, :, slice_start:slice_end
+                ]
+
+            scan = np.expand_dims(scan, axis=(0, 3))
+            template = galvo_corrections.create_refined_template(scan, x_shifts, y_shifts, key)
+
+            f = performance.parallel_motion_shifts  # function to map
+            kwargs = {
+                "raster_phase": raster_phase,
+                "fill_fraction": fill_fraction,
+                "template": template,
+            }
+            results = performance.map_frames(
```

Comment:
same as above, could you add a comment on why the field and channel are dummy inputs?